MSCL/source/mscl/Utils.cpp

```diff
@@ -471,23 +471,80 @@ namespace mscl
             strTrimRight(src);
         }
 
+        std::vector<std::string> regexFindAll(const std::string& basicString, const std::regex& pattern)
+        {
+            // Create an iterator of all the matches
+            const auto matches = std::sregex_iterator(basicString.begin(), basicString.end(), pattern);
+
+            // List to add each match
+            std::vector<std::string> matchList;
+
+            // Add each match to the list
+            for (std::sregex_iterator subMatch = matches; subMatch != std::sregex_iterator(); ++subMatch)
+            {
+                // Get the string from the match
+                std::string found = subMatch->str();
+
+                // Add it to the list to return
+                matchList.push_back(found);
+            }
+
+            return matchList;
+        }
+
+        std::vector<std::string> tokenize(const std::string& basicString, const std::string& delimiter)
+        {
+            // List to store all the delimited segments
+            std::vector<std::string> segments;
+
+            // Position tracking of delimiters
+            std::size_t previousPosition = 0;
+
+            // Find the first delimiter
+            std::size_t found;
+
+            // Extract segments until the end of the string
+            do
+            {
+                // Find the next delimiter
+                found = basicString.find(delimiter, previousPosition);
+
+                // Get the substring between the previous and current delimiter
+                const std::string substring = basicString.substr(previousPosition, found - previousPosition);
+
+                // Delimiter at the beginning/end of the string will get an empty substring
+                if (!substring.empty())
```

Comment:
Excluding empty strings can get messy if there are intentional blank elements. Shouldn't happen in the gnss receiver info, but good to be robust in util functions.

>Description: ,19.45.00
Module: 19.45.00
Fw Desc: Firmware identifier not found
Fw Version: 0.0.0